.github/workflows/changelog.yaml

```
@@ -0,0 +1,36 @@
+name: changelog entry
+
+on:
+  pull_request:
+    types: [opened, synchronize, labeled, unlabeled, reopened]
+
+permissions:
+  contents: read
+
+jobs:
+  check-changelog-entry:
+    name: Check for changelog entry
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: '0'
+
+      - name: Set up environment
+        uses: ./.github/actions/setup-env
+        with:
+          python-version: '3.9'
+          nox-setup: "tools"
+
+      - run: python -m pip install towncrier
+
+      - name: Check for presence of a Change Log fragment (only pull requests)
+        if: contains(github.event.pull_request.labels.*.name, 'skip news') != true
+        # NOTE: The pull request' base branch needs to be fetched so towncrier
+        # is able to compare the current branch with the base branch.
+        run: |
+          git fetch --no-tags origin +refs/heads/${BASE_BRANCH}:refs/remotes/origin/${BASE_BRANCH}
+          towncrier check --compare-with origin/${BASE_BRANCH}
```

```suggestion
        run: |
          if ! towncrier check --compare-with origin/${BASE_BRANCH}; then
            echo "::error::Please see https://github.com/DisnakeDev/disnake/blob/master/changelog/README.rst for details on how to create a changelog entry." >&2
            exit 1
          fi
```
Since this didn't quite work and `fetch-depth: 0` is now used, this additional fetch isn't required

for an example of the error message, see https://github.com/shiftinv/disnake/runs/7304895628?check_suite_focus=true

---

Are you sure this additional fetch isn't required? I actually don't know, but pulls from forks may require it 🤔

---

https://github.com/DisnakeDev/disnake/pull/600/commits/09ffed48789f39b935a671b92978006d597ed07b